app/javascript/controllers/fields/date_controller.js

```diff
@@ -31,6 +31,8 @@ export default class extends Controller {
   applyDateToField(event, picker) {
     const format = this.includeTimeValue ? 'MM/DD/YYYY h:mm A' : 'MM/DD/YYYY'
     $(this.fieldTarget).val(picker.startDate.format(format))
+    // bubble up a change event when the input is updated for other listeners
+    $(this.fieldTarget).trigger('change', picker)
```

Comment:
jQuery's `.trigger()` dispatches a jQuery-only event.

But, jQuery's `.on()` will catch both jQuery-only events and native browser events.

So I recommend using this instead, if you can:

```
this.fieldTarget.dispatchEvent(new Event('change'))
// we use `new Event()` and not `new CustomEvent()` because `change` is a legitimate default event for the input field.
```

But a caveat: This _might_ create an infinite loop. I'm not sure, I'd have to look into the date_controller's underlying plugin, but that underlying plugin might be listening to the change event and then doing something else, and then this `applyDateToField` will get re-triggered, again and again. In which case, a special custom event needs to be emitted and a little bit of documentation needs to be written on how to catch that special event name.

Reply:
@pascallaliberte This is what we're using downstream and so I know it works with this setup and QB. Since from the comments it looks like we're going to replace this field partial anyway (to get rid of jquery), can we merge it for now with a note to revisit it later?

Reply:
What I don't know about the pure javascript event is I need the jquery `picker` object that comes from the underlying library.

Reply:
I think it's best to do it with the native DOM event just because if we change it later you'll need to update your upstream code when we do. Up to you.

To catch the picker object on `change`, but using native DOM events, you can pass a `detail` container the `picker` object to a `CustomEvent` like this:

```
this.fieldTarget.dispatchEvent(new CustomEvent('change', { detail: { picker: picker }}))
```

And when you catch this event, the `detail` object will be available like this:

```
// manually catch a native DOM event
document.addEventListener((event) => {
  let picker = event.detail?.picker
})

// a Stimulus controller with data-action="change->some-controller#updateFormState"
updateFormState(event) {
  let picker = event.detail?.picker
}

// via jQuery
$(this).on('change', (event) => {
  let picker = event.detail?.picker
})
```